flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/rate/SnapshotGuavaRateLimiter.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.flink.cdc.connectors.mysql.rate;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import org.apache.flink.shaded.guava31.com.google.common.util.concurrent.RateLimiter;
+import org.apache.flink.util.concurrent.ExecutorThreadFactory;
+
+/**
+ * The SnapshotGuavaRateLimiter class implements Flink's RateLimiter interface, which is used to limit the read rate of data sources.
+ * It uses Guava's RateLimiter to control the number of requests allowed to pass per second,
+ * ensuring that parallel tasks can adjust the rate limit after the checkpoint is completed.
+ */
+public class SnapshotGuavaRateLimiter implements org.apache.flink.api.connector.source.util.ratelimit.RateLimiter {
+
+    private final Executor limiter =
+            Executors.newSingleThreadExecutor(new ExecutorThreadFactory("flink-snapshot-rate-limiter"));
```

Comment:
Does this rate limiter limits snapshot stage only, or both snapshot + incremental stages? The name (`SnapshotGuavaRateLimiter`) and executor name (`flink-snapshot-rate-limiter`) seems a little confusing.

Reply:
changed to `MysqlCdcGuavaRateLimiter`